pyvista/plotting/plotter.py

```diff
@@ -4567,13 +4658,15 @@ def add_silhouette(
         mapper = DataSetMapper(theme=self._theme)
         mapper.SetInputConnection(alg.GetOutputPort())
         actor, prop = self.add_actor(mapper)  # type: ignore[arg-type]
-        prop.SetColor(Color(color).float_rgb)
+        prop.SetColor(Color(color).float_rgb)  # type: ignore[call-overload]
         prop.SetOpacity(opacity)
         prop.SetLineWidth(line_width)
 
         return actor
 
-    def update_scalar_bar_range(self, clim, name=None) -> None:
+    def update_scalar_bar_range(
+        self, clim: float | tuple[float, float], name: str | None = None
```

Comment:
Is there a reason why sequence can't be used here?
```suggestion
        self, clim: float | Sequence[float, float], name: str | None = None
```

Note that we can also use `_validation.validate_data_range(clim)` to do all the validation work of checking `clim[0] < clim[1]`, checking `len(clim)==2`, and returning a `tuple[float, float]`

If we do that, then we can hint `VectorLike[float]`:

```suggestion
        self, clim: float | VectorLike[float], name: str | None = None
```

Reply:
Sequence only takes 1 type argument so that cannot be done. With regards to VectorLike[float], I think that we would like the user to know the number of elements inputted via the type checker and this doesn't allow for that.

Reply:
> Sequence only takes 1 type argument so that cannot be done. With regards to VectorLike[float], I think that we would like the user to know the number of elements inputted via the type checker and this doesn't allow for that.

We would also like to allow flexibility with the input type and not be overly constrained to only allow tuples when a list can also work... for array-like inputs we typically try to be very general with inputs and narrow the type with input validation.